DeusEx/Classes/DXRAutosave.uc

```diff
@@ -54,15 +54,16 @@ function doAutosave()
     }
 
     saveSlot = -1;
+    saveName = "DXR " $ dxr.seed $ ": " $ dxr.dxInfo.MissionLocation;
     lastMission = dxr.flags.f.GetInt('Rando_lastmission');
     if( lastMission != 0 && dxr.dxInfo.MissionNumber != 0 && lastMission != dxr.dxInfo.MissionNumber ) {
         saveSlot = 0;
+        saveName = "DXR " $ dxr.seed $ ", Mission " $ dxr.dxInfo.MissionNumber $ ": " $ dxr.dxInfo.MissionLocation;
     }
-    dxr.flags.f.SetInt('Rando_lastmission', dxr.dxInfo.MissionNumber);
+    dxr.flags.f.SetInt('Rando_lastmission', dxr.dxInfo.MissionNumber,, 999);
 
-    saveName = "DXR " $ dxr.seed $ ": " $ dxr.dxInfo.MissionLocation;
-    dxr.Player.SaveGame(saveSlot, saveName);
     bNeedSave = false;
+    dxr.Player.SaveGame(saveSlot, saveName);
```

Comment:
put the actual saving of the game after clearing the bNeedSave flag, because previously you would load the game and the flag would still be set? kinda weird that it wouldn't remember that it was inside this function though

does that mean that the interruptedDL won't get played again if you load the save? you could possibly miss a password maybe